docs/manuals/core/config/data-node-config.md

```diff
@@ -255,26 +254,82 @@ parameters can be provided.
 
 ```python linenums="1"
 from taipy import Config
+import pandas as pd
+
+def write_query_builder(data: pd.DataFrame){
+    insert_data = list(data[["date", "nb_sales"]].itertuples(index=False, name=None))
+    return [
+        "DELETE FROM forecast_table",
+        ("INSERT INTO forecast_table VALUES (?, ?)", insert_data)
+    ]
+}
 
 forecasts_cfg = Config.configure_sql_data_node(id="forecasts",
                                                db_username="admin",
                                                db_password="password",
                                                db_name="taipy",
                                                db_engine="mssql",
                                                read_query="SELECT * from forecast_table",
-                                               write_table= "forecast_table")
+                                               write_query_builder= write_query_builder)
 ```
 
 In the previous example, we configure a _sql_ data node with the id "forecasts". Its scope is the
 default value `SCENARIO`. The database username is "admin", the user's password is "password", the database name
 is "taipy", and the database engine is `mssql` (short for Microsoft SQL). The read query will be "SELECT \* from
-forecast_table", and the table the data will be written to is "forecast_table".
+forecast_table".
+The write query builder in this example is a callback that takes in a dataframe and return a list of queries. The first query will delete all the data in the table "forecast_table" and the second query is a prepared statement that takes in two values, which is the data from the two columns "date" and "nb_sales" in the dataframe. Since this is a prepared statement, it must be passed as a tuple with the first element being the query and the second element being the data.
```

Comment:
Can you add a sentence to explain that the data parameter type must be the same as the return type of the function of the task config that has the data node as output?

Reply:
In Python we do not have type in the parameter so there is not any restriction. It's up for the user to know what the return type of the task function is and code the write_query_builder accordingly.

Reply:
Yes of course, but still the data returned by the task must match with the way it is used within the query_builder. We must explain that so that the user understands it.
For instance, if the query_builder expects a data frame, the task must not return a list of integers.